src/core/lib/event_engine/posix_engine/tcp_socket_utils.h

```diff
@@ -159,12 +159,35 @@ bool SockaddrIsV4Mapped(const EventEngine::ResolvedAddress* resolved_addr,
 bool SockaddrToV4Mapped(const EventEngine::ResolvedAddress* resolved_addr,
                         EventEngine::ResolvedAddress* resolved_addr6_out);
 
-// Converts a EventEngine::ResolvedAddress into a newly-allocated human-readable
-// string.
+// Make wild card IPv6 address with specified port.
+EventEngine::ResolvedAddress SockaddrMakeWild6(int port);
+
+// Make wild card IPv4 address with specified port.
+EventEngine::ResolvedAddress SockaddrMakeWild4(int port);
+
+// Given a resolved address, return the port number in the address.
+int SockaddrGetPort(const EventEngine::ResolvedAddress& resolved_addr);
+
+// Modifes the passed address to use the specified port number. It returns true
+// if the modification operation succeeded. Otherwise it returns false. The
+// operation would only succeed if the passed address is an IPv4 or Ipv6
+// address.
+bool SockaddrSetPort(EventEngine::ResolvedAddress& resolved_addr, int port);
+
+// Unlink the path pointed to by the given address if it refers to UDS path.
+void UnlinkIfUnixDomainSocket(
+    const EventEngine::ResolvedAddress& resolved_addr);
+
+// Returns -1 if the given address is not a wildcard ipv6 or ipv6 address.
+// Otherwise returns the port number associated with the address.
+int SockaddrIsWildcard(const EventEngine::ResolvedAddress& addr);
```

Comment:
I think an `absl::optional` return type might be more appropriate here.

Reply:
Fixed.